dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/PermissionsInterface.java

```diff
@@ -182,6 +175,21 @@ static List<Permission> getOriginalOwnersForWorkflow(Workflow workflow) {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Returns the user's Google profile email if present, otherwise it returns their username. A bit
+     * of a hack to accomodate both SAM and InMemory implementations of this interface.
+     * @param user
+     * @return
+     */
+    static String emailOrUsername(User user) {
+        final User.Profile profile = user.getUserProfiles().get(TokenType.GOOGLE_COM.toString());
+        if (profile != null && profile.email != null) {
```

Comment:
I'm concerned about the base Permissions interface containing logic that depends upon the particulars of its specializations.  Would it be possible to abstract the notion of identity?  Maybe there could be a method like `String toIdentity(User user)` in the base interface, which each specialization could implement as appropriate.  The SAM implementation could map the user to an email using the Google profile, the InMemory implementation could map the user to a username, other implementations could do something else entirely...

Reply:
Good point, fixed in an upcoming push